### PR TITLE
Feature: states and error if id not found

### PR DESCRIPTION
**Pull request by DNeosphere**
States: find a state object based on the ID, abort importation

### CODE REVIEW

**Comment by drc288** [comment]
i create the api for delete and prepare the request for states

File: api/v1/views/states.py [hunk 1]
```diff
@@ -16,13 +16,21 @@ def view_states():
...
-            id_found = state.to_dict()
-            return jsonify(id_found)
-    abort(404)
+    if request.method == 'GET':
+        for state in states_obj.values():
+            if state.id == state_id:
+                id_found = state.to_dict()
+                return jsonify(id_found)
+        abort(404)
+    if request.method == 'DELETE':
```
Comment (drc288) [inline api/v1/views/states.py:41]: Update the method, create the api DELETE

**Comment by DNeosphere** [comment]
I will add the POST method to the request

File: models/city.py [hunk 2]
```diff
@@ -14,7 +14,7 @@ class City(BaseModel, Base):
         __tablename__ = 'cities'
         state_id = Column(String(60), ForeignKey('states.id'), nullable=False)
         name = Column(String(128), nullable=False)
-        places = relationship("Place", backref="cities")
+        places = relationship("Place", backref="cities", cascade="all, delete")
```
Comment (drc288) [inline models/city.py:17]: delete in cascade

**Comment by DNeosphere** [comment]
I will dop the put request

File: api/v1/views/states.py [hunk 3]
```diff
@@ -55,7 +55,7 @@ def view_state_id(state_id):
         else:
             if not request.get_json():
                 abort(404, "Not a JSON")
-            req_var = request.json()
+            req_var = request.get_json()
```
Comment (drc288) [inline api/v1/views/states.py:58]: XD

**Review by drc288 — APPROVED** [review]

**Pull request merged**